scripts/azureml-assets/azureml/assets/config.py

```diff
@@ -232,6 +232,147 @@ def release_paths(self) -> List[Path]:
         return release_paths
 
 
+class ModelType(Enum):
+    """Enum for the Model Types accepted in ModelConfig."""
+
+    MLFLOW = 'mlflow_model'
+    CUSTOM = 'custom_model'
+    TRITON = 'triton_model'
+
+
+class ModelFlavor(Enum):
+    """Enum for the Flavors accepted in ModelConfig."""
+
+    HFTRANSFORMERS = 'hftransformers'
+    PYTORCH = 'pytorch'
+
+
+class ModelTaskName(Enum):
+    """Enum for the Task names accepted in ModelConfig."""
+
+    FILL_MASK = 'fill_mask'
+    MULTICLASS = 'multiclass'
+    MULTILABEL = 'multilabel'
+    NER = 'ner'
+    QUESTION_ANSWERING = 'question-answering'
+    SUMMARIZATION = 'summarization'
+    TEXT_GENERATION = 'text-generation'
+    TEXT_CLASSIFICATION = 'text-classification'
+
+
+class ModelDownloadType(Enum):
+    """Enum for Download Method for Remote Model."""
+
+    FTP = 'ftp'
+    GIT = 'git'
+    HTTP = 'http'
+
+
+class ModelConfig(Config):
+    """Model Config class."""
+
+    """
+    Example:
+
+    path: # should contain local_path or should contain package object
+        local_path: "../models/bert-base-uncased" # the local path to the model
+        package:
+            url: https://huggingface.co/bert-base-uncased
+            commit_hash: 5546055f03398095e385d7dc625e636cc8910bf2
+            type: git_lfs
+    publish:
+        type: mlflow_model # could be one of (custom_model, mlflow_model, triton_model)
+        flavor: hftransformers # flavor should be specificed only for mlflow_model
+        task_name: translation #optional.Needed for mlflow_model huggingface flavour
+
+    """
+
+    def __init__(self, file_name: Path):
+        """Initialize object for the Model Properties extracted from extra_config model.yaml."""
+        super().__init__(file_name)
+        self._validate()
+
+    def _validate(self):
+        """Validate the yaml file."""
+        Config._validate_exists('model.path', self._path)
+        Config._validate_exists('model.publish', self._publish)
+        Config._validate_enum('model.type', self._type, ModelType, True)
+
+    @property
+    def _path(self) -> object:
+        """Model Path."""
+        return self._yaml.get('path', {})
+
+    @property
+    def path_local(self) -> Path:
+        """Model Local Path."""
+        return self._path.get('local_path')

```

Comment:
Return an actual Path object:
```suggestion
        local_path = self._path.get('local_path')
        return Path(local_path) if local_path else None
```